### PR TITLE
added dark mode option

### PR DESCRIPTION
**Pull request by iCoder-007**
With an attempt to make app more attractive I have added the dark mode to the app using theme provider since in today's time everyone like the dark mode .

Preview :

circuitversedarkmode

ScreenShots
Screenshot_20201207_142613
Screenshot_20201207_142637
Screenshot_20201207_142719

### CODE REVIEW

File: lib/ui/components/cv_header.dart [hunk 1]
```diff
@@ -27,7 +28,9 @@ class CVHeader extends StatelessWidget {
                 subtitle,
                 style: Theme.of(context).textTheme.subtitle1.copyWith(
                       fontWeight: FontWeight.bold,
-                      color: Colors.black,
+                      color: ThemeProvider.themeOf(context).id == 'dark'
```
Comment (manjotsidhu) [inline lib/ui/components/cv_header.dart:31]: I see a lot of conditional statements for switching colors from dark and white mode in multiple files. I believe we should increase reusability by moving them to a theme switching layer which will return the necessary colors in views.

**Comment by manjotsidhu** [comment]
I have tested this changes on my Android device, Dark mode is a great addition to the App but we need to work on few things. Here are some suggestions :

1. I think "Light Mode" button's place is not good to be in App Drawer, we can keep it as a Floating Action Button (near the logo) in App Drawer or somewhere better.

2. 'Label' color of fields and text color is little hard to see.


3. Image resources needs to be aligned with Dark Mode.

File: lib/main.dart [hunk 2]
```diff
@@ -30,32 +33,62 @@ class CircuitVerseMobile extends StatelessWidget {
...
+                primaryColor: PrimaryAppTheme.primaryColor,
+                accentColor: PrimaryAppTheme.primaryColor,
+                fontFamily: 'Poppins',
+                cursorColor: PrimaryAppTheme.primaryColor,
+              ),
+              description: 'LightTheme'),
+          AppTheme(
+              id: 'dark',
+              data: ThemeData(
+                primaryColor: Colors.grey[900],
```
[the remark this answers is not in the document]
Comment (iCoder-007) [reply lib/main.dart:52]: ok i will put in app_theme

File: lib/app_theme.dart [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import 'package:flutter/material.dart';
 
-class AppTheme {
-  AppTheme._();
+class PrimaryAppTheme {
```
[the remark this answers is not in the document]
Comment (iCoder-007) [reply lib/app_theme.dart:3]: I have used theme provider dependency whose AppTheme keyword was conflicting with the AppTheme class.

**Comment by jaggu21** [comment]
Adding on to @manjotsidhu 's suggestions, I think it gives better user experience if he/she can toggle between light and dark mode at any location from the app. 

Say that you're on `Add Assignments` page. You cannot toggle between light and dark modes on this page(since app drawer is not accessible), and the user will have to pop the pages from stack to reach groups page and only then the user will be able to app drawer to toggle between the two themes.

**Comment by iCoder-007** [comment]
Yes , I am working on all the changes mentioned

**Comment by iCoder-007** [comment]
I have made the asked changes

**Comment by manjotsidhu** [comment]
> I have made the asked changes

@iCoder-007 can you send screenshots for only new changes for others to review ?

**Comment by manjotsidhu** [comment]
Also please fix formatting's using `flutter format .`

**Comment by iCoder-007** [comment]
Screenshots

Screenshot_20201208_162709
Screenshot_20201208_162735
Screenshot_20201208_162812

**Comment by manjotsidhu** [comment]
I am not sure if the FAB button in home is a good idea or not. I think @Nitish145 @jaggu21 should give the inputs on this

**Comment by iCoder-007** [comment]
if you want i can place it in appbar at right

**Comment by jaggu21** [comment]
I agree with @manjotsidhu . Have a look at the following screenshot. 
FAB
Due to same color, it slightly looks off to me.

Apart from that few minor issues: 
1)Changing the color of arrow to white would make it look better than keeping it as black.
arrow
 
2) It looks better if we follow same theme throughout the entire app. Also on switching from light mode to dark mode ,the lowermost section of the HTML editor changes it's color(from grey to green) while the topmost section remains the same(grey). I think it would be better if both either change or both remain the same. Though I am not sure if this is major issue. 
t1

**Comment by iCoder-007** [comment]
so should i remove fab button and place it in appbar right
and the rest i will change to as you recommended .

**Comment by iCoder-007** [comment]
As asked i have made the changes please have a look

**Comment by manjotsidhu** [comment]
@iCoder-007 looks good to me, kindly run `flutter format .` to format the documents.

**Comment by iCoder-007** [comment]
I have already done that but everything remains unchanged

**Comment by manjotsidhu** [comment]
> I have already done that but everything remains unchanged

Here the logs say that `lib\ui\views\projects\components\project_card.dart` is not formatted
https://github.com/CircuitVerse/mobile-app/pull/40/checks?check_run_id=1523553695#step:6:87

**Comment by iCoder-007** [comment]
when i am running flutter format . it shows unchanged
and the improvements you mentioned i am fixing them

**Comment by manjotsidhu** [comment]
> when i am running flutter format . it shows unchanged

It will show Unchanged for all files except `project_card.dart` and it will format that.

**Comment by iCoder-007** [comment]
have a look

**Comment by manjotsidhu** [comment]
Here is the diff of the changes while formatting :

<details>
<summary>Diff (Click to expand)</summary>


```
diff --git a/lib/ui/views/projects/components/project_card.dart b/lib/ui/views/projects/components/project_card.dart
index 5bf414d..1a78ed7 100644
--- a/lib/ui/views/projects/components/project_card.dart
+++ b/lib/ui/views/projects/components/project_card.dart
@@ -56,8 +56,12 @@ class _ProjectCardState extends State<ProjectCard> {
               textAlign: TextAlign.center,
               style: Theme.of(context).textTheme.headline5.copyWith(
                     color: ThemeProvider.themeOf(context).id == 'dark'
-                        ? widget.isHeaderFilled ? Colors.black : Colors.white
-                        : widget.isHeaderFilled ? Colors.white : Colors.black,
+                        ? widget.isHeaderFilled
+                            ? Colors.black
+                            : Colors.white
+                        : widget.isHeaderFilled
+                            ? Colors.white
+                            : Colors.black,
:...skipping...
diff --git a/lib/ui/views/projects/components/project_card.dart b/lib/ui/views/projects/components/project_card.dart
index 5bf414d..1a78ed7 100644
--- a/lib/ui/views/projects/components/project_card.dart
+++ b/lib/ui/views/projects/components/project_card.dart
@@ -56,8 +56,12 @@ class _ProjectCardState extends State<ProjectCard> {
               textAlign: TextAlign.center,
               style: Theme.of(context).textTheme.headline5.copyWith(
                     color: ThemeProvider.themeOf(context).id == 'dark'
-                        ? widget.isHeaderFilled ? Colors.black : Colors.white
-                        : widget.isHeaderFilled ? Colors.white : Colors.black,
+                        ? widget.isHeaderFilled
+                            ? Colors.black
+                            : Colors.white
+                        : widget.isHeaderFilled
+                            ? Colors.white     
+                            : Colors.black,    
                   ),
             ),
           ),
```
</details>

**Comment by manjotsidhu** [comment]
> have a look
> 
> 

Looks like either your flutter version is old or you have set your line length more than the default.
image

**Comment by iCoder-007** [comment]
have a look at changes

**Comment by manjotsidhu** [comment]
@iCoder-007 we recently merged #37 and you might need to fix build errors and text colors for Dark Mode in Featured Project Card.

**Comment by manjotsidhu** [comment]
You will need to format `lib/ui/views/cv_landing_view.dart`.

Also fix these problems as Flutter Analyze will throw errors -
`Unused import: 'package:theme_provider/theme_provider.dart'.
Try removing the import directive.`
`Unused import: 'package:theme_provider/theme_provider.dart'.
Try removing the import directive.`

**Comment by iCoder-007** [comment]
i have resolved build errors and colors issue raised due to latest merge ,formatted the code and also executed flutter analyze with no issue found
Have a look at the changes

**Comment by manjotsidhu** [comment]
Have you fixed this arrow color ?

> 1)Changing the color of arrow to white would make it look better than keeping it as black.
> arrow

**Comment by iCoder-007** [comment]
i tried but to fix that i will have to design a custom expanded bar  ,if there is a better way please suggest me

**Comment by iCoder-007** [comment]
simply changing color of trailing icon will loose the animation

**Comment by iCoder-007** [comment]
if there is a way to detect the drawer state will solve the issue

**Comment by manjotsidhu** [comment]
@jaggu21 might help you with this issue.

**Comment by jaggu21** [comment]
@iCoder-007 [https://stackoverflow.com/questions/53584678/flutter-expansion-tile-header-color-change-and-trailing-animated-arrow-color#:~:text=For%20changing%20trailing%20arrow%20icon,dark%20and%20it%20will%20work.](https://stackoverflow.com/questions/53584678/flutter-expansion-tile-header-color-change-and-trailing-animated-arrow-color#:~:text=For%20changing%20trailing%20arrow%20icon,dark%20and%20it%20will%20work.)

This link may help you

**Comment by manjotsidhu** [comment]
@iCoder-007 there are errors coming in UI Tests because in tests we are not using ThemeProvider while pumping widgets and hence the [functions](https://github.com/iCoder-007/mobile-app/blob/introduce-dark-mode/lib/app_theme.dart#L6) that you defined for getting colors are throwing errors.

Here is the log of it -
```
2020-12-10T07:20:03.7101618Z #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
2020-12-10T07:20:03.7102655Z #1      InheritedThemeController.of (package:theme_provider/src/provider/inherited_theme.dart:19:10)
2020-12-10T07:20:03.7103726Z #2      ThemeProvider.themeOf (package:theme_provider/src/provider/theme_provider.dart:81:37)
2020-12-10T07:20:03.7104751Z #3      PrimaryAppTheme.getTextColor (package:mobile_app/app_theme.dart:13:26)
```

**Comment by iCoder-007** [comment]
i have fixed the arrow issue and also made the getting colors function independent of theme provider but to change theme dynamically theme provider dependency is still needed

**Comment by manjotsidhu** [comment]
@iCoder-007 Actually if you just add a fail-safe check that if Theme data is available then it should normally otherwise just use white mode colors.

**Comment by iCoder-007** [comment]
please check if this works fine

**Comment by manjotsidhu** [comment]
> please check if this works fine

The fail safe checks needs to also be added in the app_theme functions that you created. You can revert the use of brightness in that as well, as Theme Provider is fine. Just need to check if ThemeProvider data is available or not.

**Comment by iCoder-007** [comment]
done

> > please check if this works fine
> 
> The fail safe checks needs to also be added in the app_theme functions that you created. You can revert the use of brightness in that as well, as Theme Provider is fine. Just need to check if ThemeProvider data is available or not.

done

**Comment by manjotsidhu** [comment]
`ThemeProvider.themeOf(context)` is still throwing error in Tests.

**Comment by iCoder-007** [comment]
it might be because theme provider is also used in main.dart

**Comment by manjotsidhu** [comment]
> it might be because theme provider is also used in main.dart

Yes but tests cannot be run with ThemeProvider

**Comment by iCoder-007** [comment]
so do i need to remove the dependency

**Comment by manjotsidhu** [comment]
No, we will find a way. Did you tried `Theme.of(context)` in tests ?

**Comment by iCoder-007** [comment]
i have replaced to Theme.of(context)  I had run test and found no logs regarding theme provider , please check if this works

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
+distributionUrl=https://services.gradle.org/distributions/gradle-6.0.1-all.zip
```
Comment (manjotsidhu) [inline android/gradle/wrapper/gradle-wrapper.properties:6]: Please remove this change.

File: lib/ui/components/cv_header.dart [hunk 5]
```diff
@@ -18,17 +18,16 @@ class CVHeader extends StatelessWidget {
...
         ),
         subtitle != null
             ? Text(
                 subtitle,
                 style: Theme.of(context).textTheme.subtitle1.copyWith(
-                      fontWeight: FontWeight.bold,
-                      color: Colors.black,
-                    ),
+                    fontWeight: FontWeight.bold,
+                    color: PrimaryAppTheme.getTextColor(context)),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_header.dart:30]: Please add `,` after `(context)`and before `)` and run `flutter format` on it. It will result in better formatting's.

File: lib/ui/components/cv_password_field.dart [hunk 6]
```diff
@@ -40,17 +40,18 @@ class _CVPasswordFieldState extends State<CVPasswordField> {
         maxLines: 1,
         obscureText: _obscureText,
         keyboardType: TextInputType.visiblePassword,
-        style: TextStyle(color: Colors.black),
-        decoration: AppTheme.textFieldDecoration.copyWith(
-          suffixIcon: GestureDetector(
-            child: Icon(
-              _obscureText ? Icons.visibility_off : Icons.visibility,
-              color: AppTheme.primaryColorDark,
+        style: TextStyle(color: PrimaryAppTheme.getTextColor(context)),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_password_field.dart:43]: Better formatting's.

File: lib/ui/components/cv_password_field.dart [hunk 7]
```diff
@@ -40,17 +40,18 @@ class _CVPasswordFieldState extends State<CVPasswordField> {
...
+              ),
+              onTap: _toggle,
             ),
-            onTap: _toggle,
-          ),
-          labelText: 'Password',
-        ),
+            labelText: 'Password',
+            labelStyle:
+                TextStyle(color: PrimaryAppTheme.textfieldlabelColor(context))),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_password_field.dart:54]: Better formatting's.

File: lib/ui/components/cv_social_card.dart [hunk 8]
```diff
@@ -40,8 +42,8 @@ class CircuitVerseSocialCard extends StatelessWidget {
                       maxLines: 1,
                       overflow: TextOverflow.ellipsis,
                       style: Theme.of(context).textTheme.subtitle1.copyWith(
-                            fontWeight: FontWeight.bold,
-                          ),
+                          fontWeight: FontWeight.bold,
+                          color: PrimaryAppTheme.getTextColor(context)),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_social_card.dart:46]: Better Formatting's

File: lib/ui/components/cv_subheader.dart [hunk 9]
```diff
@@ -14,17 +15,16 @@ class CVSubheader extends StatelessWidget {
...
         ),
         subtitle != null
             ? Text(
                 subtitle,
                 style: Theme.of(context).textTheme.subtitle1.copyWith(
-                      fontWeight: FontWeight.w400,
-                      color: Colors.black,
-                    ),
+                    fontWeight: FontWeight.w400,
+                    color: PrimaryAppTheme.getTextColor(context)),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_subheader.dart:27]: Better formatting's

File: lib/ui/components/cv_text_field.dart [hunk 10]
```diff
@@ -47,10 +47,11 @@ class CVTextField extends StatelessWidget {
         maxLines: maxLines,
         keyboardType: type,
         initialValue: initialValue,
-        style: TextStyle(color: Colors.black),
-        decoration: AppTheme.textFieldDecoration.copyWith(
-          labelText: label,
-        ),
+        style: TextStyle(color: PrimaryAppTheme.getTextColor(context)),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_text_field.dart:50]: Better formatting's

File: lib/ui/components/cv_text_field.dart [hunk 11]
```diff
@@ -47,10 +47,11 @@ class CVTextField extends StatelessWidget {
...
+        decoration: PrimaryAppTheme.textFieldDecoration.copyWith(
+            labelText: label,
+            labelStyle:
+                TextStyle(color: PrimaryAppTheme.textfieldlabelColor(context))),
```
Comment (manjotsidhu) [inline lib/ui/components/cv_text_field.dart:54]: Better formatting's

File: lib/ui/views/about/about_privacy_policy_view.dart [hunk 12]
```diff
@@ -61,7 +62,8 @@ class AboutPrivacyPolicyView extends StatelessWidget {
               ? Text(
                   title,
                   style: style.copyWith(
-                      fontWeight: FontWeight.w400, color: Colors.black),
+                      fontWeight: FontWeight.w400,
+                      color: PrimaryAppTheme.primaryHeading(context)),
```
Comment (manjotsidhu) [inline lib/ui/views/about/about_privacy_policy_view.dart:66]: Better formatting's

File: lib/ui/views/about/about_privacy_policy_view.dart [hunk 13]
```diff
@@ -79,7 +81,13 @@ class AboutPrivacyPolicyView extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: showAppBar ? AppBar(title: Text('Privacy Policy')) : null,
+      appBar: showAppBar
+          ? AppBar(
+              title: Text(
+              'Privacy Policy',
+              style: TextStyle(color: PrimaryAppTheme.primaryHeading(context)),
```
Comment (manjotsidhu) [inline lib/ui/views/about/about_privacy_policy_view.dart:88]: Better formatting's

File: pubspec.lock [hunk 14]
```diff
@@ -42,28 +42,28 @@ packages:
       name: async
```
Comment (manjotsidhu) [inline pubspec.lock:42]: Remove this file from the changes.

**Review by manjotsidhu — CHANGES_REQUESTED** [review]
Please add `,` after every braces and key: value pairs, and run flutter format. This will result in better formattings.

Remove the changes done in `pubspec.lock` and `gradle-wrapper.properties`.

**Review by manjotsidhu — APPROVED** [review]
LGTM

File: lib/app_theme.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nitish145) [reply lib/app_theme.dart:3]: Can we use CVTheme or CircuitVerseTheme ? PrimaryTheme doesn't go well with the dark colours imo!

File: lib/app_theme.dart [hunk 15]
```diff
@@ -1,10 +1,78 @@
...
+  PrimaryAppTheme._();
+  static Color textfieldlabelColor(context) {
```
Comment (Nitish145) [inline lib/app_theme.dart:5]: Use camelCase to name functions.

File: lib/app_theme.dart [hunk 16]
```diff
@@ -1,10 +1,78 @@
...
+    return Theme.of(context).brightness == Brightness.dark
+        ? Colors.grey[300]
+        : Colors.grey[600];
+  }
+
+  static Color getTextColor(context) {
```
Comment (Nitish145) [inline lib/app_theme.dart:11]: better to name `textColor` for consistency among other function names involving returning colours based on theme!

File: lib/app_theme.dart [hunk 17]
```diff
@@ -1,10 +1,78 @@
...
+        : PrimaryAppTheme.bgCard;
+  }
+
+  static Color boxShadow(context) {
+    return Theme.of(context).brightness == Brightness.dark
+        ? PrimaryAppTheme.bgCardDark
+        : PrimaryAppTheme.grey;
+  }
+
+  static Color appbarText(context) {
```
Comment (Nitish145) [inline lib/app_theme.dart:35]: camelCase

File: lib/app_theme.dart [hunk 18]
```diff
@@ -1,10 +1,78 @@
...
+        : Colors.black;
+  }
+
+  static Color highlightText(context) {
+    return Theme.of(context).brightness == Brightness.dark
+        ? PrimaryAppTheme.primaryColor
+        : PrimaryAppTheme.primaryColorDark;
+  }
+
+  static Color unexpanded_trailing(context) {
```
Comment (Nitish145) [inline lib/app_theme.dart:53]: camelCase

File: lib/app_theme.dart [hunk 19]
```diff
@@ -1,10 +1,78 @@
...
+    return Theme.of(context).brightness == Brightness.dark
+        ? Colors.white
+        : Colors.black;
+  }
+
+  static Color expanded_trailing(context) {
```
Comment (Nitish145) [inline lib/app_theme.dart:59]: camelCase

File: lib/app_theme.dart [hunk 20]
```diff
@@ -1,10 +1,78 @@
...
+    return Colors.green;
+  }
+
+  static ThemeData getThemeData(context) {
```
Comment (Nitish145) [inline lib/app_theme.dart:63]: What is this function's utility? This returns the same stuff irrespective of the condition!

File: lib/app_theme.dart [hunk 21]
```diff
@@ -1,10 +1,78 @@
...
+            accentColor: Colors.green,
+            primaryColor: PrimaryAppTheme.primaryColor)
+        : ThemeData.dark().copyWith(
+            accentColor: Colors.green,
+            primaryColor: PrimaryAppTheme.primaryColor);
+  }
 
   static const Color primaryColor = Color.fromRGBO(66, 185, 131, 1);
   static const Color primaryColorDark = Color.fromRGBO(2, 110, 87, 1);
+  static const Color primaryColorDarkTheme = Color.fromRGBO(33, 33, 33, 1);
```
Comment (Nitish145) [inline lib/app_theme.dart:75]: Why is color named as theme ?

**Review by Nitish145 — CHANGES_REQUESTED** [review]

**Comment by iCoder-007** [comment]
okk

**Comment by manjotsidhu** [comment]
Also, the file names should match with class names. For example, `PrimaryAppTheme` class should have the file name as `primary_app_theme.dart`

**Comment by iCoder-007** [comment]
okk, i will do that

**Comment by iCoder-007** [comment]
I have done change requested ,please have a look

**Comment by iCoder-007** [comment]
Please check

**Comment by manjotsidhu** [comment]
@iCoder-007 can you fix the conflicts

**Comment by iCoder-007** [comment]
okk,i will do that

**Comment by iCoder-007** [comment]
Please check,is there anything left

**Comment by iCoder-007** [comment]
Please have a look, if anything is left

**Comment by manjotsidhu** [comment]
@Nitish145

File: lib/main.dart [hunk 22]
```diff
@@ -30,32 +31,85 @@ class CircuitVerseMobile extends StatelessWidget {
...
+              AppTheme(
+                  id: 'dark',
+                  data: ThemeData(
+                    primaryColor: CVTheme.secondaryColor,
+                    accentColor: CVTheme.secondaryColor,
+                    fontFamily: 'Poppins',
+                    brightness: Brightness.dark,
+                    cursorColor: CVTheme.primaryColor,
+                  ),
+                  description: 'DarkTheme')
```
Comment (Nitish145) [inline lib/main.dart:57]: Better formatting, add comma here.

File: lib/main.dart [hunk 23]
```diff
@@ -30,32 +31,85 @@ class CircuitVerseMobile extends StatelessWidget {
...
+            themes: [
+              AppTheme(
+                  id: 'light',
+                  data: ThemeData(
+                    primaryColor: CVTheme.primaryColor,
+                    accentColor: CVTheme.primaryColor,
+                    fontFamily: 'Poppins',
+                    cursorColor: CVTheme.primaryColor,
+                  ),
+                  description: 'LightTheme'),
```
Comment (Nitish145) [inline lib/main.dart:47]: Better formatting, add comma here.

**Review by Nitish145 — CHANGES_REQUESTED** [review]

**Comment by iCoder-007** [comment]
made the requested changes

File: lib/cv_theme.dart [hunk 24]
```diff
@@ -0,0 +1,86 @@
...
+  static const Color primaryColorLight = Color.fromRGBO(66, 185, 131, 0.5);
+  static const Color primaryColorShadow = Color.fromRGBO(245, 255, 252, 1);
+  static const Color imageBackground = Color.fromRGBO(63, 61, 86, 1);
+  static const Color secondaryColor = Color.fromRGBO(33, 33, 33, 1);
+  static const Color blue = Color.fromRGBO(66, 185, 182, 1);
+  static const Color red = Color.fromRGBO(255, 89, 89, 1);
+  static const Color grey = Color.fromRGBO(150, 150, 150, 1);
+  static const Color lightGrey = Color.fromRGBO(150, 150, 150, 0.5);
+  static const Color bgCard = Color.fromRGBO(255, 255, 255, 0.9);
+  static Color bgCardDark = Colors.grey[700];
```
Comment (Nitish145) [inline lib/cv_theme.dart:67]: Mark these as const. Const has its own benefits and overall consistency matters.

File: lib/cv_theme.dart [hunk 25]
```diff
@@ -0,0 +1,86 @@
...
+  static Color htmlEditorBg = Colors.grey[100];
```
Comment (Nitish145) [inline lib/cv_theme.dart:68]: Same as above

File: lib/main.dart [hunk 26]
```diff
@@ -30,32 +31,87 @@ class CircuitVerseMobile extends StatelessWidget {
...
+            onGenerateRoute: CVRouter.generateRoute,
+            theme: ThemeData(
+              primarySwatch: generateMaterialColor(CVTheme.primaryColor),
+              fontFamily: 'Poppins',
+              cursorColor: CVTheme.primaryColor,
+            ),
+            home: StartUpView(),
+          );
   }
 }
```
Comment (Nitish145) [inline lib/main.dart:117]: Code repetition for `GetMaterialApp` makes it look messy. Can be refactored.

**Review by Nitish145 — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 26 above)
Comment (iCoder-007) [reply lib/main.dart:117]: i added so that ui test doesn't fail

Comment (manjotsidhu) [reply lib/main.dart:117]: @iCoder-007 If you don't use condition then UI Tests fails ?

Comment (iCoder-007) [reply lib/main.dart:117]: yes that is why i added it

Comment (manjotsidhu) [reply lib/main.dart:117]: I just now removed this condition and using `ThemeProvider` code, the UI Tests are passing.

<details>
<summary> Here is the diff </summary>

```
diff --git a/lib/main.dart b/lib/main.dart
index 3c7c4cb..c6161d9 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -31,8 +31,7 @@ class CircuitVerseMobile extends StatelessWidget {
       DeviceOrientation.portraitDown,
     ]);
 
-    return ThemeProvider != null
-        ? ThemeProvider(
+    return ThemeProvider(
             saveThemesOnChange: true,
             loadThemeOnInit: true,
             themes: [
@@ -85,33 +84,6 @@ class CircuitVerseMobile extends StatelessWidget {
                 ),
               ),
             ),
-          )
-        : GetMaterialApp(
-            title: 'CircuitVerse Mobile',
-            localizationsDelegates: [
-              AppLocalizationsDelegate(),
-              GlobalMaterialLocalizations.delegate,
-              GlobalWidgetsLocalizations.delegate,
-            ],
-            supportedLocales: [
-              Locale('en', ''),
-            ],
-            onGenerateTitle: (BuildContext context) =>
-                AppLocalizations.of(context).title,
-            debugShowCheckedModeBanner: false,
-            builder: (context, child) => Navigator(
-              key: locator<DialogService>().dialogNavigationKey,
-              onGenerateRoute: (settings) => MaterialPageRoute(
-                builder: (context) => DialogManager(child: child),
-              ),
-            ),
-            onGenerateRoute: CVRouter.generateRoute,
-            theme: ThemeData(
-              primarySwatch: generateMaterialColor(CVTheme.primaryColor),
-              fontFamily: 'Poppins',
-              cursorColor: CVTheme.primaryColor,
-            ),
-            home: StartUpView(),
           );
   }
 }

```

</details>

Comment (iCoder-007) [reply lib/main.dart:117]: so should i remove the condition

Comment (manjotsidhu) [reply lib/main.dart:117]: Yes

Comment (manjotsidhu) [reply lib/main.dart:117]: Also test the changes on your device once.

Comment (iCoder-007) [reply lib/main.dart:117]: I just tried the ui test after removing condition it shows all test passes

**Comment by iCoder-007** [comment]
I have done the changes as asked , have a look

**Comment by manjotsidhu** [comment]
@Nitish145

File: lib/ui/views/groups/my_groups_view.dart [hunk 27]
```diff
@@ -85,9 +86,8 @@ class _MyGroupsViewState extends State<MyGroupsView> {
             Text(
               'Groups',
               style: Theme.of(context).textTheme.headline4.copyWith(
-                    fontWeight: FontWeight.bold,
-                    color: Colors.black,
-                  ),
+                  fontWeight: FontWeight.bold,
+                  color: CVTheme.textColor(context)),
```
Comment (manjotsidhu) [inline lib/ui/views/groups/my_groups_view.dart:90]: Formatting here

**Review by manjotsidhu — APPROVED** [review]

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 28]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
```
Comment (Nitish145) [inline android/gradle/wrapper/gradle-wrapper.properties:6]: Unnecessary changes. Can u please remove these.

File: lib/main.dart (hunk 26 above)
Comment (Nitish145) [reply lib/main.dart:79]: Looks far better now. Nice.

File: lib/ui/views/groups/my_groups_view.dart (hunk 27 above)
Comment (Nitish145) [reply lib/ui/views/groups/my_groups_view.dart:90]: @iCoder-007

File: test/ui_tests/home/home_view_test.dart [hunk 29]
```diff
@@ -9,7 +9,6 @@ import 'package:mobile_app/ui/views/home/home_view.dart';
 import 'package:mobile_app/utils/router.dart';
 import 'package:mockito/mockito.dart';
 import 'package:shared_preferences/shared_preferences.dart';
-
```
Comment (Nitish145) [inline test/ui_tests/home/home_view_test.dart:12]: This line is the distinction between relative and absolute imports. Prefer not to remove it.

**Review by Nitish145 — CHANGES_REQUESTED** [review]

**Comment by iCoder-007** [comment]
Please check

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 28 above)
Comment (Nitish145) [reply android/gradle/wrapper/gradle-wrapper.properties:6]: @iCoder-007

**Review by Nitish145 — CHANGES_REQUESTED** [review]

**Comment by Nitish145** [comment]
@iCoder-007 can u please undo the last merge commit, and try rebasing to fix the conflicts

**Comment by iCoder-007** [comment]
i saw that that there are some new changes in master branch so i have pulled the changes and will fix the conflicts manually , will this works

**Comment by Nitish145** [comment]
Yes

**Comment by iCoder-007** [comment]
I think due to latest merge the ui test are failing

**Comment by iCoder-007** [comment]
And sorry for the delays  I was quiet busy

**Comment by manjotsidhu** [comment]
@iCoder-007 Have you merged https://github.com/CircuitVerse/mobile-app/commit/d60561b595726a82b44af311f4e0f672516a8c6e ?

**Comment by manjotsidhu** [comment]
@iCoder-007 try pushing again, the CI will pass

**Comment by iCoder-007** [comment]
okk

File: pubspec.lock (hunk 14 above)
Comment (Nitish145) [reply pubspec.lock:42]: @iCoder-007

**Review by Nitish145 — CHANGES_REQUESTED** [review]
Please make the requested changes.

**Comment by manjotsidhu** [comment]
@iCoder-007 Can you check if `CVTypeAheadField` which was merged is following the dark theme as well ?

**Comment by iCoder-007** [comment]
> @iCoder-007 Can you check if `CVTypeAheadField` which was merged is following the dark theme as well ?

yes, and sorry for delay

**Comment by manjotsidhu** [comment]
@Nitish145

**Comment by iCoder-007** [comment]
please check

**Comment by manjotsidhu** [comment]
@tachyons Seems like @Nitish145 is busy, can you merge this ?
